src/legacy/ui/ui_render/ui_render_mixin.js

```diff
@@ -95,7 +95,7 @@ export function uiRenderMixin(osdServer, server, config) {
       const darkMode =
         !authEnabled || request.auth.isAuthenticated
           ? await uiSettings.get('theme:darkMode')
-          : false;
+          : uiSettings.getOverrideOrDefault('theme:darkMode');
```

Comment:
This is fine, but I'd kind of expect a call to `uiSettings.get()` to handle all the value resolution internally - that it's just an interface to fetch the one true value, not a more narrowly-limited API for getting an explicitly set value. But I may very well be misunderstanding the overall architectural design of this plugin and systems.

Reply:
`uiSettings.get()` calls `getAll` who calls `getRaw` who calls `read` with no params. However `read` is capable of dealing with 401s if asked to:
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/b125f12c2ba1191b4697965a2fcb15256b4873ba/src/core/server/ui_settings/ui_settings_client.ts#L229-L232

However, the blast radius of changing this default is unknown at this point. We should look at this at a later point in time. Maybe, just give `get`, `getAll`, and `getRaw` and optional `ignore401Errors` to pass down, but we will have to invest in that separately.

Reply:
Yeah, I definitely don't have enough context to have a strong opinion, and best to defer.